database/src/main/java/com/firebase/ui/database/FirebaseRecyclerAdapter.java

```diff
@@ -63,32 +62,23 @@
  *         }
  *     };
  *     recycler.setAdapter(mAdapter);
- * }
- * </pre></blockquote>
+ * </pre>
  *
- * @param <T> The Java class that maps to the type of objects stored in the Firebase location.
+ * @param <T>  The Java class that maps to the type of objects stored in the Firebase location.
  * @param <VH> The ViewHolder class that contains the Views in the layout that is shown for each object.
  */
 public abstract class FirebaseRecyclerAdapter<T, VH extends RecyclerView.ViewHolder> extends RecyclerView.Adapter<VH> {
 
-    Class<T> mModelClass;
     protected int mModelLayout;
+    Class<T> mModelClass;
     Class<VH> mViewHolderClass;
     FirebaseArray mSnapshots;
 
-    /**
-     * @param modelClass Firebase will marshall the data at a location into an instance of a class that you provide
-     * @param modelLayout This is the layout used to represent a single item in the list. You will be responsible for populating an
-     *                    instance of the corresponding view with the data from an instance of modelClass.
-     * @param viewHolderClass The class that hold references to all sub-views in an instance modelLayout.
-     * @param ref        The Firebase location to watch for data changes. Can also be a slice of a location, using some
-     *                   combination of <code>limit()</code>, <code>startAt()</code>, and <code>endAt()</code>
-     */
-    public FirebaseRecyclerAdapter(Class<T> modelClass, int modelLayout, Class<VH> viewHolderClass, Query ref) {
+    FirebaseRecyclerAdapter(Class<T> modelClass, int modelLayout, Class<VH> viewHolderClass, FirebaseArray snapshots) {
```

Comment:
@puf With this implementation, should `FirebaseRecyclerAdapter(Class<T> modelClass, int modelLayout, Class<VH> viewHolderClass, FirebaseArray snapshots)` be protected or public? I'm keeping it private to the library because it's more a private API to support `IndexFirebaseArray` than anything else. Any thoughts?